src/components/Tool/TntConnect/TntConnect.tsx

```diff
@@ -0,0 +1,392 @@
+}));
+
+const BottomBox = styled(Box)(({ theme }) => ({
+  backgroundColor: theme.palette.cruGrayLight.main,
+  width: '100%',
+  padding: theme.spacing(1),
+  display: 'flex',
+  justifyContent: 'end',
+  [theme.breakpoints.down('sm')]: {
+    paddingTop: theme.spacing(2),
+  },
+}));
+
+const OuterBox = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'center',
+  width: '100%',
+  padding: theme.spacing(3),
+}));
+
+const BorderBox = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+  marginBottom: theme.spacing(2),
+  border: `1px solid ${theme.palette.cruGrayMedium.main}`,
+  width: '100%',
+}));
+
+const Section = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+}));
+
+const VisuallyHiddenInput = styled('input')({
+  clip: 'rect(0 0 0 0)',
+  clipPath: 'inset(50%)',
+  height: 1,
+  overflow: 'hidden',
+  position: 'absolute',
+  bottom: 0,
+  left: 0,
+  whiteSpace: 'nowrap',
+  width: 1,
+});
+
+const useStyles = makeStyles()(() => ({
+  bulletList: {
+    margin: '0 0 10px 15px',
+    '> li': {
+      margin: '5px',
+    },
+  },
+  divider: {
+    margin: theme.spacing(2, 0),
+  },
+}));
+
+const tntSchema = yup.object({
+  selectedTags: yup.array().of(yup.string()).default([]),
+  override: yup.string().required(),
+});
+
+type Attributes = yup.InferType<typeof tntSchema>;
+
+interface Props {
+  accountListId: string;
+}
+
+const TntConnect: React.FC<Props> = ({ accountListId }: Props) => {
+  const { classes } = useStyles();
+  const { appName } = useGetAppSettings();
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: sessionData } = useSession();
+  const [showModal, setShowModal] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [tntFile, setTntFile] = useState<{
+    file: File;
+    blobUrl: string;
+  } | null>(null);
+
+  const { data: contactTagsList, loading: contactTagsListLoading } =
+    useGetContactTagListQuery({
+      variables: {
+        accountListId,
+      },
+    });
+
+  const handleCloseModal = () => {
+    setShowModal(false);
+    window.location.href = `${process.env.SITE_URL}/accountLists/${accountListId}/tools`;
+    setLoading(true);
+  };
+
+  const onSubmit = async (
+    attributes: Attributes,
+    { resetForm },
+  ): Promise<void> => {
+    const file = tntFile?.file;
+    if (file) {
+      try {
+        await uploadTnt({
+          override: attributes.override,
+          selectedTags: attributes.selectedTags,
+          file,
+          t,
+          apiToken: sessionData?.user?.apiToken || '',
+          accountListId,
+        });
+      } catch (err) {
+        enqueueSnackbar(
+          err instanceof Error ? err.message : t('File could not be uploaded'),
+          {
+            variant: 'error',
+          },
+        );
+        return;
+      }
+    }
+    resetForm();
+    setTntFile(null);
+    enqueueSnackbar(t('Upload Complete'), {
```

Comment:
Do you want to mention something about how their upload is being processed and give it time for MPDX to update?

Reply:
The modal pops up and explains that it will take time for their upload to process.